components/capi/include/pluralrules.h

```diff
@@ -0,0 +1,46 @@
+// This file is part of ICU4X. For terms of use, please see the file
+// called LICENSE at the top level of the ICU4X source tree
+// (online at: https://github.com/unicode-org/icu4x/blob/main/LICENSE ).
+
+#pragma once
+#include <stdint.h>
+#include <stddef.h>
+#include <stdbool.h>
+#include "provider.h"
+#include "locale.h"
+
+// opaque
+typedef struct PluralRules PluralRules; 
+
+typedef struct {
+    PluralRules* rules;
+    bool success;
+} CreatePluralRulesResult;
+
+typedef enum {
+    PluralRuleType_Cardinal,
+    PluralRuleType_Ordinal
+} PluralRuleType;
+
+typedef enum {
+    PluralCategory_Zero,
+    PluralCategory_One,
+    PluralCategory_Two,
+    PluralCategory_Few,
+    PluralCategory_Many,
+    PluralCategory_Other,
+} PluralCategory;
+
+typedef struct {
+    uint64_t i;
+    size_t v;
+    size_t w;
+    uint64_t f;
+    uint64_t t;
+    size_t c;
+} PluralOperands;
+
+
+CreatePluralRulesResult plural_rules_create(Locale* locale, ErasedDataProvider* provider, PluralRuleType ty);
```

Comment:
ICU4C uses "open" and "close" instead of "create" and "destroy".  In ICU4C, this function is called `uplrules_open()`.  Should we use the same convention?

```suggestion
CreatePluralRulesResult icu4x_plural_rules_open(Locale* locale, ErasedDataProvider* provider, PluralRuleType ty);
```

Reply:
If this always allocates memory, I think create/destroy (or new/free) is clearer. To me open/close implies there might be some kind of reference counting scheme going on in the background.

Reply:
Open makes it sound like we're opening a file, to me. `_create` and `_destroy` are pretty standard in FFI land